core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabaseAdmin.java

```diff
@@ -1,73 +1,186 @@
+          ImmutableMap.<RdbEngine, ImmutableMap<DataType, String>>builder()
+              .put(
+                  RdbEngine.MYSQL,
+                  ImmutableMap.<DataType, String>builder()
+                      .put(DataType.TEXT, "VARCHAR(64)")
+                      .put(DataType.BLOB, "VARBINARY(64)")
+                      .build())
+              .put(
+                  RdbEngine.POSTGRESQL,
+                  ImmutableMap.<DataType, String>builder()
+                      .put(DataType.TEXT, "VARCHAR(10485760)")
+                      .build())
+              .put(
+                  RdbEngine.ORACLE,
+                  ImmutableMap.<DataType, String>builder()
+                      .put(DataType.TEXT, "VARCHAR2(64)")
+                      .put(DataType.BLOB, "RAW(64)")
+                      .build())
+              .put(RdbEngine.SQL_SERVER, ImmutableMap.<DataType, String>builder().build())
+              .build();
   private final BasicDataSource dataSource;
-  private final Optional<String> namespacePrefix;
+  private final Optional<String> schemaPrefix;
   private final JdbcTableMetadataManager metadataManager;
+  private final RdbEngine rdbEngine;
 
   @Inject
   public JdbcDatabaseAdmin(JdbcConfig config) {
     dataSource = JdbcUtils.initDataSource(config);
-    namespacePrefix = config.getNamespacePrefix();
-    RdbEngine rdbEngine = JdbcUtils.getRdbEngine(config.getContactPoints().get(0));
-    metadataManager = new JdbcTableMetadataManager(dataSource, namespacePrefix, rdbEngine);
+    schemaPrefix = config.getNamespacePrefix();
+    rdbEngine = JdbcUtils.getRdbEngine(config.getContactPoints().get(0));
+    metadataManager = new JdbcTableMetadataManager(dataSource, schemaPrefix, rdbEngine);
   }
 
   @VisibleForTesting
   public JdbcDatabaseAdmin(
-      JdbcTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
-    dataSource = null;
+      BasicDataSource dataSource,
+      JdbcTableMetadataManager metadataManager,
+      Optional<String> namespacePrefix,
+      RdbEngine rdbEngine) {
+    this.dataSource = dataSource;
     this.metadataManager = metadataManager;
-    this.namespacePrefix = namespacePrefix.map(n -> n + "_");
+    this.schemaPrefix = namespacePrefix;
+    this.rdbEngine = rdbEngine;
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    try (Connection connection = dataSource.getConnection()) {
+      connection.setAutoCommit(false);
+      connection.setTransactionIsolation(Connection.TRANSACTION_SERIALIZABLE);
+      try {
+        createSchemaIfNotExists(connection, namespace);
+        createTableInternal(connection, namespace, table, metadata);
+        createIndex(connection, namespace, table, metadata);
+      } catch (SQLException e) {
+        connection.rollback();
```

Comment:
Just a question.
Are we supposed to call `rollback` for DDL?
Here, it's called, but it seems like it is not called in other places like below.
https://github.com/scalar-labs/scalardb/pull/270/files#diff-26e58dccc9f536e4794af12538f00972936415bb04ba1b0d5d1f6b903f55e2d5R169

Are there any particular criteria when to call it and when to not?

Reply:
```
Are we supposed to call rollback for DDL?
```

> Are we supposed to call rollback for DDL?

I assumed rollback was supported for all kind of statements but it's not the case for DDL. Oracle and Mysql do not support DDL statement transaction rollback while Postgresql and Sqlserver do. 

> Here, it's called, but it seems like it is not called in other places like below.
> https://github.com/scalar-labs/scalardb/pull/270/files#diff-26e58dccc9f536e4794af12538f00972936415bb04ba1b0d5d1f6b903f55e2d5R169
> 
> Are there any particular criteria when to call it and when to not?

`connection.rollback()` should only be called  if `connection.setAutoCommit(false)` was called beforehand. 
By default, auto-commit is true so every statement executed will be auto-committed right after. If an errors occurs in this mode, it will be roll-backed transparently.

Reply:
Got it. Thank you!